packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.utils
+
+import com.android.build.api.variant.AndroidComponentsExtension
+import com.facebook.react.utils.PropertyUtils.INTERNAL_DISABLE_JAVA_VERSION_ALIGNMENT
+import org.gradle.api.Action
+import org.gradle.api.JavaVersion
+import org.gradle.api.Project
+import org.gradle.api.plugins.AppliedPlugin
+
+internal object JdkConfiguratorUtils {
+  /**
+   * Function that takes care of configuring the JDK toolchain for all the projects projects. As we
+   * do decide the JDK version based on the AGP version that RNGP brings over, here we can safely
+   * configure the toolchain to 11.
+   */
+  fun configureJavaToolChains(input: Project) {
+    if (input.hasProperty(INTERNAL_DISABLE_JAVA_VERSION_ALIGNMENT)) {
```

Comment:
nit: it's hasProperty so `react.internal.disableJavaVersionAlignment=false` will also disable the java version enforcement. but it doesn't really matter though 😅

Reply:
doesn't have permission to approve the pr, but it looks good to me. thanks @cortinico.

Reply:
> it's hasProperty so react.internal.disableJavaVersionAlignment=false will also disable the java version enforcement. but it doesn't really matter though 😅

Yeah I prefer this approach as the property is called `disable...` so we won't have to handle all the boolean cases (true/True/1/etc). Plus it's an `.internal.` property so no need to be super strict here